assets/src/scripts/components/hydrograph/points.js

```diff
@@ -37,25 +54,37 @@ const lineSegmentsSelector = memoize(tsDataKey => createSelector(
         // Accumulate data into line groups, splitting on the estimated and
         // approval status.
         let lines = [];
+
         let lastClasses = {};
 
-        for (let pt of points) {
-            // Ignored masked data
-            if (pt.value === null) {
-                lastClasses = {};
-                continue;
-            }
+        const masks = new Set(Object.keys(MASK_DESC));
 
+        for (let pt of points) {
             // Classes to put on the line with this point.
-            const lineClasses = {
+            let lineClasses = {
                 approved: pt.approved,
-                estimated: pt.estimated
+                estimated: pt.estimated,
+                dataMask: null
             };
-
+            if (pt.value === null) {
+                let qualifiers = new Set(pt.qualifiers.map(q => q.toLowerCase()));
+                // current business rules specify that a particular data point
+                // will only have at most one masking qualifier
+                let maskIntersection = new Set([...masks].filter(x => qualifiers.has(x)));
+                lineClasses.dataMask = [...maskIntersection][0]
+            }
+            else if (pt.value !== null) {
```

Comment:
This can just be an else {}

Reply:
Yep.

Reply:
Now that I look at it more closely, I think that's a merge error...